File: client/src/app/data/types/Response.ts
import { EventType } from './EventType';
import { GameEntry } from './GameEntry';
import { MessageData } from './MessageData';
import { Scope } from './Scope';
import { Winner } from './Winner';

export interface Response {
  scope: Scope;
  eventType: EventType;
  data: GameEntry | MessageData | Winner | string;
}

File: pubsub/Model/Response.cs
using System.Text.Json.Serialization;

namespace PubSub.Model;

# nullable enable

public class Response
{
  [JsonConverter(typeof(JsonStringEnumConverter))]
  [JsonPropertyName("scope")]
  public Scope Scope { get; set; }

  [JsonConverter(typeof(JsonStringEnumConverter))]
  [JsonPropertyName("eventType")]
  public EventType EventType { get; set; }

  [JsonPropertyName("data")]
  public dynamic? Data { get; set; }
}